src/Workspaces/Core/Portable/SymbolKey/SymbolKey.FunctionPointerTypeSymbolKey.cs

```diff
@@ -0,0 +1,36 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis
+{
+    internal partial struct SymbolKey
+    {
+        private static class FunctionPointerTypeSymbolKey
+        {
+            public static void Create(IFunctionPointerTypeSymbol symbol, SymbolKeyWriter visitor)
+            {
+                visitor.WriteRefKind(symbol.Signature.RefKind);
+                visitor.WriteSymbolKey(symbol.Signature.ReturnType);
+                visitor.WriteRefKindArray(symbol.Signature.Parameters);
+                visitor.WriteParameterTypesArray(symbol.Signature.Parameters);
+            }
+
+            public static SymbolKeyResolution Resolve(SymbolKeyReader reader)
+            {
+                var returnRefKind = reader.ReadRefKind();
+                var returnType = reader.ReadSymbolKey();
+                using var paramRefKinds = reader.ReadRefKindArray();
+                using var paramTypes = reader.ReadSymbolKeyArray<ITypeSymbol>();
```

Comment:
i can't remember the semantics of ReadSymbolKeyArray, does it fail if any of them fail?  or does it just toss that value.  if the latter, you hve to check that the lengths of these two arrays are the same.  (which is pretty sane to do anyways).

Reply:
It fails if any of them fail.